config/environment.go

```diff
@@ -26,14 +26,36 @@ func GetEnvironment() map[string]string {
 }
 
 func applyEnvKey(key, value string, rValueSubject reflect.Value) {
+	var rFieldValue reflect.Value
 	keyParts := strings.SplitN(key, "_", 2)
 	if len(keyParts) < 1 {
 		return
 	}
-	rFieldValue := rValueSubject.FieldByNameFunc(func(candidate string) bool {
-		candidateUpper := strings.ToUpper(candidate)
-		return candidateUpper == keyParts[0]
-	})
+	if rValueSubject.Kind() == reflect.Struct {
```

Comment:
Are we sure this is always a struct if it's not a map? I'm concerned we could be too restrictive here

Reply:
function `FieldByNameFunc` fails on non structured subjects, which means this is more safe way to do it.